### PR TITLE
feat: More ways to specify R path

### LINKED ISSUE #61: Define path to R executable
Hello, for me it would useful if I could somehow define the path to the R executable on my machine instead of using the RPATH environment variable.

Would it be possible to do something similar to the Python version where it looks for the path definied in the Python Extension? Instead looking for the path definined in the R Extension using the following record in setting.json.
`"r.rpath.windows":`

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #61

Adds support for consulting the IDE for the preferred R runtime in two ways:

1. In Positron, we now consult the preffered runtime, i.e. the runtime selected by the user for the project. The extension will follow the prefferred runtime version on launch, meaning that if you launch the app with R 4.4, change to R 4.2, and relaunch the app, the second launch will use R 4.2.

2. In VS Code, the [R Debugger extension](https://marketplace.visualstudio.com/items?itemName=RDebugger.r-debugger) includes `r.rpath.{windows,mac,linux}` settings. We also consult this setting. Note that the R Debugger extension should not be used in Positron. We don't skip this check if in Positron (maybe you're syncing settings?), though.

The R Path consultation order is now:

1. Positron
2. `r.rpath.{windows,mac,linux}` setting
3. `$PATH`
4. Windows registry

### CODE REVIEW

File: src/extension-api-utils/extensionHost.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return (url: string) => pst.window.previewUrl(vscode.Uri.parse(url));
+}
+
+export async function getPositronPreferredRuntime(languageId: string): Promise<LanguageRuntimeMetadata | undefined> {
+    const pst = getPositronAPI();
+    if (!pst) { return; }
+    return await pst.runtime.getPreferredRuntime(languageId);
+}
+
+function getPositronAPI(): void | any {
```
Comment (jcheng5) [inline src/extension-api-utils/extensionHost.ts:48]: Feels weird to treat "void" as a type like that. I think this is more idiomatic.
```suggestion
function getPositronAPI(): undefined | any {
```

File: src/extension-api-utils/extensionHost.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    if (!(typeof globalThis === "object")) { return; }
+    if (!("acquirePositronApi" in globalThis)) { return; }
+    if (typeof globalThis.acquirePositronApi !== "function") { return; }
+
+    return globalThis.acquirePositronApi();
```
Comment (jcheng5) [inline src/extension-api-utils/extensionHost.ts:53]: Just out of curiosity, not asking for a change... Is this all equivalent to `globalThis.acquirePositronApi?()`? (Other than you're gracefully dealing with `globalThis` not being an object and `acquirePositronApi` not being a function, both of which seem unlikely?)

**Review by jcheng5 — APPROVED** [review]
Other than the one type tweak, LGTM!

File: src/extension-api-utils/extensionHost.ts (hunk 2 above)
Comment (gadenbuie) [reply src/extension-api-utils/extensionHost.ts:53]: You mean `globalThis?.acquirePositronApi()`, right? Then yes, it's equivalent if overly cautious. I'm happy to switch to the cleaner syntax.

Comment (jcheng5) [reply src/extension-api-utils/extensionHost.ts]: Hmmm, I don't think that's right... I think `globalThis?.` checks for `globalThis` being null/undefined but not `acquirePositronApi`. It looks to me like `?()` is not a thing, I think maybe the original is best.

Or
```typescript
const apa = globalThis?.acquirePositronApi;
return apa ? apa() : undefined;
```

Or
```typescript
if (typeof acquirePositronApi === "function") {
  return acquirePositronApi();
}
```

(Bikeshedding so hard right now... sorry... :grimacing:)

Comment (jcheng5) [reply src/extension-api-utils/extensionHost.ts]: Or

```typescript
globalThis?.acquirePositronApi?.call(globalThis)
```
(Alright I'm done, I promise!)

Comment (gadenbuie) [reply src/extension-api-utils/extensionHost.ts]: > I don't think that's right... I think `globalThis?.` checks for `globalThis` being null/undefined but not `acquirePositronApi`

Oh yeah, that's right. I think I'd rather revert back to something closer to what I had originally:

```js
function getPositronAPI(): undefined | any {
    if (typeof globalThis?.acquirePositronApi !== "function") { return; }

    return globalThis.acquirePositronApi();
}
```

**Pull request merged**